### PR TITLE
Merging #1253 (Remove check if access_token is not returned for id_token only scenarios)

### PR DESCRIPTION
**Pull request by oldalton**
## Proposed changes

Merging a PR from an external contributor: #1253

The PR Azure Pipelines checks are not working for forked repos, so opening a new PR to get PR checks completed.

## Type of change

- [ ] Feature work
- [ ] Bug fix
- [ ] Documentation
- [x] Engineering change
- [ ] Test
- [ ] Logging/Telemetry

## Risk

- [ ] High – Errors could cause MAJOR regression of many scenarios. (Example: new large features or high level infrastructure changes)
- [x] Medium – Errors could cause regression of 1 or more scenarios. (Example: somewhat complex bug fixes, small new features)
- [ ] Small – No issues are expected. (Example: Very small bug fixes, string changes, or configuration settings changes)

### CODE REVIEW

File: MSAL/src/MSALResult.m [hunk 1]
```diff
@@ -135,10 +135,6 @@ + (MSALResult *)resultWithMSIDTokenResult:(MSIDTokenResult *)tokenResult
     }
```
Comment (kaisong1990) [inline MSAL/src/MSALResult.m:135]: This pull request does not update CHANGELOG.md.

Please consider if this change would be noticeable to a partner or user and either update CHANGELOG.md or resolve this conversation.

File: MSAL/src/MSALResult.m [hunk 2]
```diff
@@ -135,10 +135,6 @@ + (MSALResult *)resultWithMSIDTokenResult:(MSIDTokenResult *)tokenResult
...
     
     NSString *accessToken = [authScheme getClientAccessToken:tokenResult.accessToken popManager:popManager error:error];
```
Comment (hieunguyenmsft) [inline MSAL/src/MSALResult.m:137]: To really distinguish the case no accessToken vs parsing accessToken error here, I think we should change to:
if (tokenResult.accessToken is not empty)
{
    // parsing access token
}  else {
// move on to the next step.
}

**Review by hieunguyenmsft — CHANGES_REQUESTED** [review]

File: MSAL/src/public/MSALResult.h [hunk 3]
```diff
@@ -40,13 +40,13 @@
 #pragma mark - Token response
 
 /** The Access Token requested. */
-@property (readonly, nonnull) NSString *accessToken;
+@property (readonly, nullable) NSString *accessToken;
```
Comment (kaisong1990) [inline MSAL/src/public/MSALResult.h:43]: Bit concerned here as they were nonnulll earlier and could cause crashes without null check

File: MSAL/src/MSALResult.m (hunk 2 above)
Comment (oldalton) [reply MSAL/src/MSALResult.m]: Addressed, thanks.

File: MSAL/src/public/MSALResult.h (hunk 3 above)
Comment (oldalton) [reply MSAL/src/public/MSALResult.h:43]: Do you have any suggestions? We can return an empty string, but that feels unnatural.

**Review by kaisong1990 — APPROVED** [review]

File: MSAL/src/public/MSALResult.h (hunk 3 above)
Comment (kaisong1990) [reply MSAL/src/public/MSALResult.h]: As discussed offline, we can resolve this

**Review by hieunguyenmsft — APPROVED** [review]
lgtm

**Pull request merged**